### ISSUE TITLE
tests for error scenarios

### ISSUE THREAD

**Issue by gabemontero** [shipwright-io/operator#22]
See the discussion starting in https://github.com/shipwright-io/operator/pull/14#discussion_r718695602

we currently do not have testing that sorts out that the operator does the right thing if something is amiss with the shipwright build controller deployment

we'll take any combination of unit, integration, or e2e's as a starting point to resolve this, where if need be, follow up issues are opened if further investigation warrants additional tests

among other things, we should see what other OLM controllers like tekton, or even what openshift CVO operators do in their unit tests to validate bad inputs

**Comment by otaviof** [comment]
Currently, this operator only generates and applies a Build-Controller `Deployment` object. If the object is invalid, this operator will surface the error, but the work is done when the deployment is declared.

I think what we are about to discuss is related to the `operator` watching over the Build-Controller deployment status, and also surface the errors in case the deployment does not reach "ready" status after some time.

Does it make sense to you, folks?

**Comment by gabemontero** [comment]
yes your clarification here is warranted @otaviof .... that is really what I was getting after with my comments in https://github.com/shipwright-io/operator/pull/14#discussion_r718695602

thanks for the corrections / tweaks for the intent here

**Comment by adambkaplan** [comment]
I think there are a few bugs open which are related to this issue:

- #50 
- #47
